document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
 
 * TODO: Relationship between original file encoding, offset, and UTF-8 byte sequence.
 * TODO: Explain responsibility of viewer/editor to match encoding.
+
+## `notification` object { #notification }
+
+### General { #notification-General }
+
+A `notification` object describes a condition encountered in the course of running an analysis tool
+which is relevant to the operation of the tool itself,
+as opposed to being relevant to a file being analyzed by the tool.
+Conditions relevant to files being analyzed by a tool are represented by `result` objects (&sect;[#result]).
+
+### `id` property { #notification-id }
+
+A `notification` object may contain a property named `id` whose value is a string containing
+the stable, opaque identifier for the condition that was encountered.
+
+### `ruleId` property { #notification-ruleId }
```

Comment:
I see. So is a notification happens to be raised in some context that relates directly to a rule, we provide a placeholder for that data. Yes. Solves a known issue we've encountered in production (rule 'foo' crashed, for example.). OK, agree